### PR TITLE
OK-2629 add CI

### PR DESCRIPTION
**Pull request by relwell**
Adds CI for pull requests that check the following:

* formatting
* dependencies
* passing tests

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -27,6 +27,10 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version: 1.21
+      - name: Format code
```
Comment (spikeburton) [inline .github/workflows/release.yml:30]: Do we need to run the format step in the release workflow?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: orka-go-ci
+
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (spikeburton) [inline .github/workflows/pr.yml:6]:
```suggestion
on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
```

Usually we add a trigger for push to the main branch as well ... so something like this

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: orka-go-ci
```
Comment (spikeburton) [inline .github/workflows/pr.yml:1]: We should rename the workflow, right? 😜

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        run: |
+          make tidy
+          git diff --exit-code
+
+      - name: Format code
+        run: |
+          make fmt
+          git diff --exit-code
+
+      - name: Check for unused dependencies
```
Comment (spikeburton) [inline .github/workflows/pr.yml:32]: Looks like we have this step twice (already above on line 22)

File: .github/workflows/release.yml (hunk 1 above)
Comment (relwell) [reply .github/workflows/release.yml:30]: I suppose note, but it does assume no one has made any mistakes on `main`

File: .github/workflows/pr.yml (hunk 4 above)
Comment (relwell) [reply .github/workflows/pr.yml:35]: Whoops, thanks

**Comment by relwell** [comment]
@spikeburton should be ready to roll

**Review by spikeburton — APPROVED** [review]

**Pull request merged**